nuxeo-common/src/test/java/org/nuxeo/common/utils/TestI18NUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Estelle Giuly <egiuly@nuxeo.com>
+ */
+
+package org.nuxeo.common.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.util.Locale;
+
+import org.junit.Test;
+import org.nuxeo.common.utils.i18n.I18NUtils;
+
+public class TestI18NUtils {
+
+    @Test
+    public void testParseLocaleWhenOnlyLanguage() {
+        Locale parsedLocale = I18NUtils.parseLocale("it");
+        assertEquals(new Locale("it"), parsedLocale);
+    }
+
+    @Test
+    public void testParseLocaleWhenLanguageAndCountry() {
+        Locale parsedLocale = I18NUtils.parseLocale("en_US");
+        assertEquals(new Locale("en", "US"), parsedLocale);
+    }
+
+    @Test
+    public void testParseLocaleWhenLanguageCountryAndVariant() {
+        Locale parsedLocale = I18NUtils.parseLocale("th_TH_TH");
+        assertEquals(new Locale("th", "TH", "TH"), parsedLocale);
+    }
+
+    @Test
+    public void testParseLocaleWhenAlpha3Code() {
+        Locale parsedLocale = I18NUtils.parseLocale("FRE_FRA");
+        assertEquals(new Locale("fre", "FRA"), parsedLocale);
+    }
+
+    @Test
+    public void testParseLocaleWhenDashes() {
+        Locale parsedLocale = I18NUtils.parseLocale("en-US");
+        assertEquals(new Locale("en", "US"), parsedLocale);
+    }
+
+    @Test
+    public void testParseLocaleWhenExcessiveLodash() {
+        Locale parsedLocale = I18NUtils.parseLocale("en_");
+        assertEquals(new Locale("en"), parsedLocale);
+    }
+
+    @Test
+    public void testParseLocaleWhenNull() {
+        try {
+            I18NUtils.parseLocale(null);
+            fail();
+        } catch (NullPointerException e) {
+            // ok
```

Comment:
i'd rather avoid the NPE here and return a null locale (same if the localeString is "")